src/nvim/terminal.c

```diff
@@ -169,6 +169,54 @@ static VTermScreenCallbacks vterm_screen_callbacks = {
 
 static Set(ptr_t) invalidated_terminals = SET_INIT;
 
+static void emit_term_request(void **argv)
+{
+  char *payload = argv[0];
+  size_t payload_length = (size_t)argv[1];
+
+  String termrequest = { .data = payload, .size = payload_length };
+  Object data = STRING_OBJ(termrequest);
+  set_vim_var_string(VV_TERMREQUEST, payload, (ptrdiff_t)payload_length);
+  apply_autocmds_group(EVENT_TERMREQUEST, NULL, NULL, false, AUGROUP_ALL, curbuf, NULL, &data);
+  xfree(payload);
+}
+
+static int on_osc(int command, VTermStringFragment frag, void *user)
+{
+  if (frag.str == NULL) {
+    return 0;
+  }
+
+  StringBuilder request = KV_INITIAL_VALUE;
+  kv_printf(request, "\x1b]%d;", command);
+  kv_concat_len(request, frag.str, frag.len);
+  multiqueue_put(main_loop.events, emit_term_request, request.items, (void *)request.size);
+  return 1;
+}
+
+static int on_dcs(const char *command, size_t commandlen, VTermStringFragment frag, void *user)
+{
+  if ((command == NULL) || (frag.str == NULL)) {
+    return 0;
+  }
+
+  StringBuilder request = KV_INITIAL_VALUE;
+  kv_printf(request, "\x1bP%*s", (int)commandlen, command);
```

Comment:
coverity warnings, do we need to assert?

```
*** CID 477623:  Null pointer dereferences  (FORWARD_NULL)
/src/nvim/terminal.c: 205 in on_dcs()
199       if ((command == NULL) || (frag.str == NULL)) {
200         return 0;
201       }
202     
203       StringBuilder request = KV_INITIAL_VALUE;
204       kv_printf(request, "\x1bP%*s", (int)commandlen, command);
>>>     CID 477623:  Null pointer dereferences  (FORWARD_NULL)
>>>     Passing null pointer "request.items + request.size" to "memcpy", which dereferences it.
205       kv_concat_len(request, frag.str, frag.len);
206       multiqueue_put(main_loop.events, emit_term_request, request.items, (void *)request.size);
207       return 1;
208     }
209     
210     static VTermStateFallbacks vterm_fallbacks = {
```

Reply:
Interesting, I don't think `request` should ever be `NULL` after `kv_printf`. Surely we don't need to `assert` after every `kv_printf` call?

Reply:
> Surely we don't need to `assert` after every `kv_printf` call?

i think coverity is looking at this and thinking `kv_printf` could return early: https://github.com/neovim/neovim/blob/2dc439c672facaeb8e51ef6aa20efc0e7092eee2/src/nvim/strings.c#L2144

we could add an assert in `kv_concat_len` ?

Reply:
https://github.com/neovim/neovim/pull/27022